cilium-cli/cli/connectivity_test.go

```diff
@@ -104,3 +105,79 @@ func TestConnectivityTestFlags(t *testing.T) {
 	require.NoError(t, ct.Flags().Set("node-selector", "c=d"))
 	require.Equal(t, map[string]string{"a": "b", "c": "d"}, params.NodeSelector)
 }
+
+func TestPrintImageArtifacts(t *testing.T) {
+	ct := newCmdConnectivityTest(&api.NopHooks{})
+	var buf bytes.Buffer
+
+	// Test print-image-artifacts flag for connectivity test subcommand
+	buf.Reset()
+	ct.SetOut(&buf)
+	require.NoError(t, ct.Flags().Set("print-image-artifacts", "true"))
+	require.NoError(t, ct.Execute())
+	expectedImages := []string{
+		defaults.ConnectivityCheckAlpineCurlImage,
+		defaults.ConnectivityCheckJSONMockImage,
+		defaults.ConnectivityDNSTestServerImage,
+		defaults.ConnectivityTestConnDisruptImage,
+		defaults.ConnectivityTestFRRImage,
+		defaults.ConnectivityTestSocatImage,
+	}
+	for _, img := range expectedImages {
+		require.Contains(t, buf.String(), img)
+	}
+
+	// Test print-image-artifacts flag for connectivity test subcommand with overridden image
+	buf.Reset()
+	ct.SetOut(&buf)
+	var alpineImage = "alpine/curl:latest"
+	require.NoError(t, ct.Flags().Set("print-image-artifacts", "true"))
+	require.NoError(t, ct.Flags().Set("curl-image", alpineImage))
+	require.NoError(t, ct.Execute())
+	expectedImagesWithOverride := []string{
+		alpineImage,
+		defaults.ConnectivityCheckJSONMockImage,
+		defaults.ConnectivityDNSTestServerImage,
+		defaults.ConnectivityTestConnDisruptImage,
+		defaults.ConnectivityTestFRRImage,
+		defaults.ConnectivityTestSocatImage,
+	}
+	for _, img := range expectedImagesWithOverride {
+		require.Contains(t, buf.String(), img)
+	}
+
+	// Ensure no images are printed when the flag is not set
+	buf.Reset()
+	ct.SetOut(&buf)
+	require.NoError(t, ct.Flags().Set("print-image-artifacts", "false"))
+	require.NoError(t, ct.Execute())
```

Comment:
Here we need to make sure that we don't run any tests which would require a running cluster, so I think we'll need to `ct.Flags().Set("test", "!.*")` or similar to exclude all tests.